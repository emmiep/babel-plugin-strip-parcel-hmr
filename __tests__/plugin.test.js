const plugin = require('../src/plugin');
const {transformFileSync} = require('babel-core');
const path = require('path');

const defaultBabelOptions = {
  babelrc: false
};

const fixturesDir = path.join(__dirname, '__fixtures__');

describe('plugin', () => {
  let babelOptions;

  beforeEach(() => babelOptions = Object.assign({}, defaultBabelOptions, {
    plugins: [plugin]
  }));

  it('removes standalone HMR test if-clauses', () => {
    const fixture = fixturePath('remove-standalone-if');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('extracts else-statements following HMR test if-clauses', () => {
    const fixture = fixturePath('extract-following-else');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('extracts else-if-statements following HMR test if-clauses', () => {
    const fixture = fixturePath('extract-following-else-if');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('combines if-clauses surrounding HMR test if-statements', () => {
    const fixture = fixturePath('combine-surrounding-if');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('ignores other properties of the HMR object', () => {
    const fixture = fixturePath('ignore-other-object-properties');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('ignores other objects', () => {
    const fixture = fixturePath('ignore-other-objects');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('ignores negated HMR tests', () => {
    const fixture = fixturePath('ignore-negated-tests');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('ignores local variable bindings', () => {
    const fixture = fixturePath('ignore-local-variables');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });

  it('allows customizing object and property names', () => {
    babelOptions = {
      plugins: [
        [plugin, {
          objectName: 'customObjectName',
          propertyName: 'customPropertyName'
        }]
      ]
    };

    const fixture = fixturePath('use-custom-names');
    const output = transformFileSync(fixture, babelOptions);
    expect(output.code).toMatchSnapshot();
  });
});

function fixturePath(name) {
  return path.join(fixturesDir, `${name}.js`);
}

