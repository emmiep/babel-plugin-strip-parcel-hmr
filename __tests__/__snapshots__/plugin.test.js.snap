// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin when transforming conditional expressions ignores local variable bindings 1`] = `
"const module = { hot: true };

const value = module.hot ? shouldBeKept : shouldAlsoBeKept;"
`;

exports[`plugin when transforming conditional expressions ignores other object properties 1`] = `"const value = module.notHot ? shouldBeKept : shouldAlsoBeKept;"`;

exports[`plugin when transforming conditional expressions ignores other objects 1`] = `"const value = notModule.hot ? shouldBeKept : shouldAlsoBeKept;"`;

exports[`plugin when transforming conditional expressions substitutes expressions with matching negated tests 1`] = `"const value = shouldBeKept;"`;

exports[`plugin when transforming conditional expressions substitutes expressions with matching tests 1`] = `"const value = shouldBeKept;"`;

exports[`plugin when transforming conditional expressions substitutes expressions with matching tests using computed string properties 1`] = `"const value = shouldBeKept;"`;

exports[`plugin when transforming conditional expressions substitutes expressions with matching tests using in operators 1`] = `"const value = shouldBeKept;"`;

exports[`plugin when transforming conditional expressions substitutes expressions with matching tests using typeof operators 1`] = `"const value = shouldBeKept;"`;

exports[`plugin when transforming if statements combines if-clauses surrounding matching if-clauses 1`] = `
"if (module.notHot) {
  shouldBeKept;
} else {
    shouldAlsoBeKept;
  }"
`;

exports[`plugin when transforming if statements extracts else-if-statements following matching if-clauses 1`] = `
"if (module.notHot) {
  shouldBeKept;
} else {
  shouldAlsoBeKept;
}"
`;

exports[`plugin when transforming if statements extracts else-statements following matching if-clauses 1`] = `
"{
  shouldBeKept;
}"
`;

exports[`plugin when transforming if statements extracts standalone matching negated if-clauses 1`] = `
"{
  shouldBeExtracted;
}"
`;

exports[`plugin when transforming if statements ignores local variable bindings 1`] = `
"const module = { hot: true };

if (module.hot) {
  shouldBeKept;
}"
`;

exports[`plugin when transforming if statements ignores other object properties 1`] = `
"if (module.notHot) {
  shouldBeKept;
}"
`;

exports[`plugin when transforming if statements ignores other objects 1`] = `
"if (notModule.hot) {
  shouldBeKept;
}"
`;

exports[`plugin when transforming if statements removes standalone matching if-clauses 1`] = `
"before;

after;"
`;

exports[`plugin when transforming if statements removes standalone matching if-clauses using computed string properties 1`] = `
"before;

after;"
`;

exports[`plugin when transforming if statements removes standalone matching if-clauses using in operators 1`] = `
"before;

after;"
`;

exports[`plugin when transforming if statements removes standalone matching if-clauses using typeof operators 1`] = `
"before;

after;"
`;

exports[`plugin when using options allows customizing object and property names 1`] = `
"if (module.hot) {
  shouldBeKept;
}"
`;
